proxy/test/proxy.test.ts

```diff
@@ -0,0 +1,163 @@
+import { equal } from 'node:assert'
+import { createServer } from 'node:http'
+import { before, describe, test } from 'node:test'
+import { URL } from 'node:url'
+
+import createProxyServer from '../proxy.js'
+import { getTestHttpServer, initTestHttpServer } from './utils.js'
+
+/**
+ * Proxy is tested with a service that outputs request and response data.
+ *
+ * <local client> -> <proxy> -> <targetServer>
+ */
+const targetServer = createServer((req, res) => {
+  let parsedUrl = new URL(req.url || '', `http://${req.headers.host}`)
+
+  let queryParams = Object.fromEntries(parsedUrl.searchParams.entries())
+  let requestPath = parsedUrl.pathname
+
+  res.writeHead(200, {
+    'Content-Type': 'text/json'
+  })
+
+  res.end(
+    JSON.stringify({
+      request: {
+        headers: req.headers,
+        method: req.method,
+        queryParams,
+        requestPath,
+        url: req.url
+      },
+      response: 'ok'
+    })
+  )
+})
+
+describe('proxy tests', async () => {
```

Comment:
We don’t use single `describe` per file in other tests.

Reply:
Can we have a single describe here? The reason is that I want to utilise `before` and `after` so that I can init and destroy http server. 

If I ditch single describe, then I'd have to write a lot of:

```js
await initTestHttpServer(
    'proxy',
    createProxyServer({ hostnameWhitelist: ['localhost'], silentMode: true })
  )
  await initTestHttpServer('target', targetServer)

  let proxyServerUrl = ''
  let targetServerUrl = ''

  before(() => {
    let proxy = getTestHttpServer('proxy')
    if (proxy) {
      proxyServerUrl = proxy.baseUrl
    }

    let target = getTestHttpServer('target')
    if (target) {
      targetServerUrl = target.baseUrl
    }

    if (!target || !proxy) {
      throw new Error(
        "Couldn't set up target server or proxy server. Please check out 'proxy.test.js'"
      )
    }
  })
```

Maybe there is a way to support before outside the single describe?

Reply:
You can use `before`/`after` without `description`.

Just use `beforeEach` or `beforeAll` from `node:test`.